train.py

```diff
@@ -259,12 +266,38 @@ def train(self, model, train_dataloader, eval_dataloader, val_dict, report=False
                     loss.backward()
                     optim.step()
 
+                    if self.discriminator is not None:
+                        # As in the paper, we're going to train the discriminator exactly the same number of times as the model, once after every epoch.
```

Comment:
We can tune this later but I was wondering if we should experiment with putting this discriminator training step in a for loop (run it k times for every time the qa model has a parameter update). that value of 'k' would also be a hyperparam